x-pack/plugins/reporting/server/lib/screenshots/get_screenshots.ts

```diff
@@ -25,6 +27,38 @@ export const getScreenshots = async (
   for (let i = 0; i < elementsPositionAndAttributes.length; i++) {
     const endTrace = startTrace('get_screenshots', 'read');
     const item = elementsPositionAndAttributes[i];
+
+    // In Puppeteer 5.4+, the viewport size limits what the screenshot can take, even if a clip is specified. The clip area must
+    // be visible in the viewport. This workaround resizes the viewport to the actual content height and width.
+    // NOTE: this will fire a window resize event
+
+    // Check current viewport size
+    const [height, width] = await browser.evaluate(
+      {
+        fn: () => [document.body.clientHeight, document.body.clientWidth],
+        args: [],
+      },
+      { context: 'resize for screenshot' },
+      logger
+    );
+
+    logger.debug(`Browser viewport: height: ${height}, width: ${width}`);
+
+    // Resize the viewport if the clip area is not visible
```

Comment:
For `print` layout, this is not ideal. I didn't see issues while testing locally, but I worry about repeatedly increasing the viewport as it captures one panel after the other.